src/tplr/comms.py

```diff
@@ -780,149 +294,74 @@ async def get(
-        window: int,
-        key: str,
-        timeout: int,
-        local: bool = True,
-        stale_retention: int = 10,
-        time_min: datetime = None,
-        time_max: datetime = None,
-    ) -> Optional[dict]:
-        """GET with retry operation."""
-        start_time = time.time()
-        end_time = start_time + timeout
-        tried_after_time_max = False
-        time_max_grace_period = 3.0
-
-        while True:
-            # Check if we've timed out
-            if time.time() >= end_time:
-                tplr.logger.debug(f"GET {uid}/{window}/{key} timed out.")
-                return None
-
-            # Check if we're past time_max with grace period
-            now = datetime.now(timezone.utc)
-
-            # Only consider it "past time_max" if we're 3 seconds beyond time_max
-            past_time_max = False
-            if time_max is not None and now > time_max:
-                seconds_past_time_max = (now - time_max).total_seconds()
-                past_time_max = seconds_past_time_max > time_max_grace_period
-
-            # If we're past time_max (with grace period) and already tried once, don't retry again
-            if past_time_max and tried_after_time_max:
-                tplr.logger.debug(
-                    f"Already tried once after time_max + {time_max_grace_period}s for UID {uid}, window {window}. Stopping retries."
-                )
-                return None
-
-            # If we're past time_max (with grace period), mark that we've tried once
-            if past_time_max:
-                tried_after_time_max = True
-                tplr.logger.debug(
-                    f"Past time_max + {time_max_grace_period}s for UID {uid}, window {window}. This is the final retry."
-                )
-
-            # Make the request
-            state_dict = await self.get(
-                uid=uid,
-                window=window,
-                key=key,
-                local=local,
-                stale_retention=stale_retention,
-                time_min=time_min,
-                time_max=time_max,
-            )
-
-            # Check for TOO_LATE/TOO_EARLY markers
-            if isinstance(state_dict, dict):
-                if state_dict.get("__status") == "TOO_LATE":
-                    tplr.logger.info(
-                        f"Gradient for UID {uid}, window {window} exists but was uploaded too late. Skipping."
-                    )
-                    return None
-                elif state_dict.get("__status") == "TOO_EARLY":
-                    tplr.logger.info(
-                        f"Gradient for UID {uid}, window {window} exists but was uploaded too early. Skipping."
-                    )
-                    return None
-
-            # If we got a result, return it
-            if state_dict is not None:
-                return state_dict
-
-            # Short delay before retrying
-            await asyncio.sleep(0.1)
-
     async def gather(
```

Comment:
`key` and `stale_retention` is unused.

Reply:
also not sure if we need `my_uid` but need to remove it in the gradietn manager as well then.